File: app/src/main/java/com/example/socialapp/network/RetrofitClient.java
package com.example.socialapp.network;


import static com.example.socialapp.utils.Constants.FACEBOOK_GRAPH_URL;

import com.example.socialapp.network.livedataadapter.LiveDataCallAdapterFactory;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static RetrofitClient instance = null;
    private Api api;
    HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();


    private RetrofitClient() {
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();

        Retrofit retrofitAuth = new Retrofit.Builder()
                .baseUrl(FACEBOOK_GRAPH_URL)
                .client(client)
                .addCallAdapterFactory(new LiveDataCallAdapterFactory())
                .addConverterFactory(GsonConverterFactory.create())

                .build();
        api = retrofitAuth.create(Api.class);

    }
    public static synchronized RetrofitClient getInstance() {
        if (instance == null) {
            instance = new RetrofitClient();
        }
        return instance;
    }

    public Api getApi() {

        return api;
    }

}

File: app/src/main/java/com/example/socialapp/network/RetrofitClient.kt
package com.example.socialapp.network

import okhttp3.logging.HttpLoggingInterceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import com.example.socialapp.network.livedataadapter.LiveDataCallAdapterFactory
import com.example.socialapp.utils.Constants
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient  {
    val api: Api
    private var interceptor = HttpLoggingInterceptor()
    init {
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
        val client: OkHttpClient = OkHttpClient.Builder().addInterceptor(interceptor).build()
        val retrofitAuth = Retrofit.Builder()
            .baseUrl(Constants.FACEBOOK_GRAPH_URL)
            .client(client)
            .addCallAdapterFactory(LiveDataCallAdapterFactory())
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        api = retrofitAuth.create(Api::class.java)
    }
}
